[Visual Refresh] Update docs to reflect new tokens

mgadewoll opened the pull request:
## Summary

>[!IMPORTANT]
This PR merges into a feature branch.

Closes https://github.com/elastic/eui-private/issues/174

This PR updates the EUI and EUI+ docs to ensure the new and/or updated color tokens are reflected.

### Changes

- adds output and docs for additional new tokens (text tokens, background tokens, border tokens)
- updates descriptions that mention computed color values, as Borealis is not generating colors anymore
- removes docs for color utilities that are deprecated (not expected to be promoted anymore)
- updates examples of color contrast to add additional color token options
- [fix]: updates brand color values for DARK mode - aligning it with design (I'm not sure why this was wrong, it might have been wrongly updated in a rebase 🫠)
- [cleanup]: updates (deprecated) token usages where new semantic tokens are available (e.g. `dangerText` -> `textDanger` or `text` -> `textParagraph`)

## QA

- [x] review the EUI and EUI+ docs
  - [x] text tokens ([EUI](https://eui.elastic.co/pr_8306/#/theming/colors/values#text-colors), [EUI+](https://eui.elastic.co/pr_8306/new-docs/docs/theming/colors/#text-colors))
  - [x] background tokens ([EUI](https://eui.elastic.co/pr_8306/#/theming/colors/values#background-colors), [EUI+](https://eui.elastic.co/pr_8306/new-docs/docs/theming/colors/#background-colors))
  - [x] border tokens ([EUI](https://eui.elastic.co/pr_8306/#/theming/colors/values#border-colors), [EUI+](https://eui.elastic.co/pr_8306/new-docs/docs/theming/colors/#border-colors))
  - [x] color utilities ([EUI](https://eui.elastic.co/pr_8306/#/theming/colors/utilities), [EUI+](https://eui.elastic.co/pr_8306/new-docs/docs/theming/colors/utilities/))
  - [x] color contrast ([EUI](https://eui.elastic.co/pr_8306/#/theming/colors/contrast))
  - [x] theme customization ([EUI](https://eui.elastic.co/pr_8306/#/theming/customizing-themes))

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8306/
- Storybook: https://eui.elastic.co/pr_8306/storybook
- New docs: https://eui.elastic.co/pr_8306/new-docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3400)
* Commit: 40e99eafb94e4b20212253f70aaa909c27b6839e

cc @mgadewoll

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3400","number":3400,"commit":"40e99eafb94e4b20212253f70aaa909c27b6839e"}],"number":3400}
buildkite-pr-comment-->

tkajtoch reviewed (approved):
The changes are looking solid! 🚢

acstll commented:
Thank you @mgadewoll! I only have 2 non-blocking comments regarding text:

- the first paragraph in the Colors page has this statement (screenshot), I think it's no longer true, is it?
- the Shades and Special colors are deprecated in the current site but not the new
- extra: in the tokens table, I think it would be nice if the `Sample` for backgrounds and borders was not `Aa` but a "square" (this is maybe something for a new issue for EUI+?)

![df68e3e3723764e6f3b47373dea8229d](https://github.com/user-attachments/assets/654e64ab-b920-4fab-9dca-8293d6649d70)

mgadewoll commented:
> Thank you @mgadewoll! I only have 2 non-blocking comments regarding text:
> 
> * the first paragraph in the Colors page has this statement (screenshot), I think it's no longer true, is it?
> * the Shades and Special colors are deprecated in the current site but not the new
> * extra: in the tokens table, I think it would be nice if the `Sample` for backgrounds and borders was not `Aa` but a "square" (this is maybe something for a new issue for EUI+?)
> 
> ![df68e3e3723764e6f3b47373dea8229d](https://private-user-images.githubusercontent.com/851105/409859912-654e64ab-b920-4fab-9dca-8293d6649d70.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg3MzkwOTAsIm5iZiI6MTczODczODc5MCwicGF0aCI6Ii84NTExMDUvNDA5ODU5OTEyLTY1NGU2NGFiLWI5MjAtNGZhYi05ZGNhLTgyOTNkNjY0OWQ3MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwNVQwNjU5NTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNDE1Yzc0YThhZDc3MjgwNWVlYzYxODAwNzUwMmU0NGNlNjc0OGVhMWMyNmZmN2UwZjUwNGY1YzRlOTgwZmIwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.yZUStwhRWuULAVBj2n5ZNE--gJOo8r2tpV--8g7l0gk)

>* the first paragraph in the Colors page has this statement (screenshot), I think it's no longer true, is it?

True! I missed that one 🫠 
I'll update to conditionally switch in current EUI docs, as the text is still correct for Amsterdam. But I think we can update EUI+ docs to use Borealis-only wording. Wdyt?
Updated in this [commit](https://github.com/elastic/eui/pull/8306/commits/06b76d469e1e8df8cbf828e5112920ed192840bc).

> * the Shades and Special colors are deprecated in the current site but not the new
True! I missed updating the description as it's not actually using the JSDoc string anymore 🤦‍♀️ Thanks for the catch!
Updated in this [commit](https://github.com/elastic/eui/pull/8306/commits/515464c5cde301835de75e87c3171157dae8ece5).

> * extra: in the tokens table, I think it would be nice if the `Sample` for backgrounds and borders was not `Aa` but a "square" (this is maybe something for a new issue for EUI+?)

Ah yes, good call! 👍 
Updated in this [commit](https://github.com/elastic/eui/pull/8306/commits/5d5feeb995679158c806ae8de5c8fc657a3f512b).

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8306/
- Storybook: https://eui.elastic.co/pr_8306/storybook
- New docs: https://eui.elastic.co/pr_8306/new-docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3413)
* Commit: 515464c5cde301835de75e87c3171157dae8ece5

### History
* :green_heart: [Build #3400](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3400) succeeded 40e99eafb94e4b20212253f70aaa909c27b6839e

cc @mgadewoll

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3413","number":3413,"commit":"515464c5cde301835de75e87c3171157dae8ece5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3400","number":3400,"commit":"40e99eafb94e4b20212253f70aaa909c27b6839e"}],"number":3413}
buildkite-pr-comment-->

acstll commented:
> But I think we can update EUI+ docs to use Borealis-only wording. Wdyt?

👌
